src/main/java/CurlLoggingFilter.java

```diff
@@ -0,0 +1,98 @@
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.FilterConfig;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.stereotype.Component;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.Enumeration;
+
+@Component
+public class CurlLoggingFilter implements Filter {
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        // Initialization logic, if needed.
+    }
+
+    @Override
+    public void doFilter(jakarta.servlet.ServletRequest request, jakarta.servlet.ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+
+        HttpServletRequest httpRequest = (HttpServletRequest) request;
+        HttpServletResponse httpResponse = (HttpServletResponse) response;
+
+        // Wrap the request so that the body can be read multiple times
+        CachedBodyHttpServletRequest cachedRequest = new CachedBodyHttpServletRequest(httpRequest);
+
+        // Build the curl command from the request
+        String curlCommand = buildCurlCommand(cachedRequest);
+
+        // Log the curl command to the console
+        System.out.println(curlCommand);
+
+        // Continue the filter chain
+        chain.doFilter(cachedRequest, response);
+    }
+
+    @Override
+    public void destroy() {
+        // Cleanup logic, if needed.
+    }
+
+    private String buildCurlCommand(HttpServletRequest request) throws IOException {
+        StringBuilder curl = new StringBuilder("curl -X ");
+        curl.append(request.getMethod()).append(" '").append(request.getRequestURL());
+
+        String queryString = request.getQueryString();
+        if (queryString != null) {
+            curl.append('?').append(queryString);
+        }
+        curl.append("'");
+
+        // Add headers
+        Enumeration<String> headerNames = request.getHeaderNames();
+        while (headerNames.hasMoreElements()) {
+            String headerName = headerNames.nextElement();
+            String headerValue = request.getHeader(headerName);
+            curl.append(" -H '").append(headerName).append(": ").append(headerValue).append("'");
```

Comment:
Iterating over parameters using Enumeration is outdated. Consider using modern constructs like 'request.getParameterMap()' instead.